core/src/state/wallet.rs

```diff
@@ -283,7 +285,34 @@ where
 impl From<Wallet> for SizedCircuitWallet {
     fn from(wallet: Wallet) -> Self {
         // Recover the wallet from its secret shares to maintain ordering for witnesses
```

Comment:
This comment is no longer really true, right? Now we just recover the wallet from the plaintext elements stored on it

To that end, why did we recover from private/public shares previously, if we already store the elements in plaintext?

Reply:
Yeah thanks, removed the comment. The reason was to preserve ordering of balances, orders, fees; between the output of this method and the on-chain state. 

We need to preserve ordering so that `VALID REBLIND` proves correctly (elements of the reblinded wallet shouldn't be re-ordered). 

The indexed state filters out default orders, balances, etc for API convenience, but this pushes all the non-default orders to the front of the recovered wallet, which doesn't necessarily reflect the in-contract secret share state. Practically, it's simpler to implement `From` this way and just reconstruct the witness to `VALID REBLIND` directly from the known shares when we need ordering.